spec/lib/content_exporter_spec.rb

```diff
@@ -0,0 +1,199 @@
+      allow(I18n).to receive(:t).with("results_link") { broken_result_fixture }
+      expect { results_rows }.to raise_error(KeyError, "key not found: :id")
+    end
+
+    it "returns raises an error when a link does not have a text value" do
+      broken_result_fixture = { group_one: { subgroup_one: { title: "Title", items: [{ id: "0001" }] } } }
+      allow(I18n).to receive(:t).with("results_link") { broken_result_fixture }
+      expect { results_rows }.to raise_error(KeyError, "key not found: :text")
+    end
+
+    it "returns raises an error when a link does not have a group.title value" do
+      broken_group_fixture = { group_one: {} }
+      allow(I18n).to receive(:t).with("coronavirus_form.groups") { broken_group_fixture }
+      expect { results_rows }.to raise_error(KeyError, "key not found: :title")
+    end
+
+    it "returns raises an error when a link does not have a subgroup :title value" do
+      broken_result_fixture = { group_one: { subgroup_one: { items: [{ id: "0001", text: "bloop" }] } } }
+      allow(I18n).to receive(:t).with("results_link") { broken_result_fixture }
+      expect { results_rows }.to raise_error(KeyError, "key not found: :title")
+    end
+  end
+
+  describe "#generate_results_link_csv" do
+    csv_fixture = "id,group_title,subgroup_title,support_and_advice,text,href,show_to_nations\n" \
+      "0001,I am the title for Group one,I am the title for Group one subgroup one,false,\"This is a row that only has text, it will appaer like a paragraph\",\"\",\"\"\n" \
+      "0002,I am the title for Group one,I am the title for Group one subgroup one,false,\"This is a row has text and an href, it will appaer as an anchor tag\",http://test.stubbed.gov.uk,\"\"\n" \
+      "0003,I am the title for Group one,I am the title for Group one subgroup one,false,\"This is a row has text, an href and group criteira, it will appear as an anchor tag if the user's answers match the criteria\",http://test.stubbed.llyw.cymru,Wales\n" \
+      "0004,I am the title for Group one,I am the title for Group one subgroup one,true,\"This is a row has text, an href and multiple group criteira, it will appear as an anchor tag if the user's answers match the more complex criteria\",http://test.stubbed.gb.gov.uk,Wales OR Scotland OR England\n" \
+      "0005,I am the title for Group one,I am the title for Group one subgroup two,false,\"This is a row that only has text, it will appaer like a paragraph\",\"\",\"\"\n" \
+      "0006,I am the title for Group two,I am the title for Group two subgroup one,false,\"This is a row that only has text, it will appaer like a paragraph\",\"\",\"\"\n"
+    let(:csv) { ContentExporter.generate_results_link_csv }
+
+    before do
+      allow(I18n).to receive(:t).with("results_link") { locale_results_link_fixture }
+      allow(I18n).to receive(:t).with("coronavirus_form.groups") { locale_form_groups_fixture }
+    end
+
+    it "outputs a csv with correct headers" do
```

Comment:
This test is redundant given the test @195.

Reply:
Yeah fair,
I think because Rspec does a string match here you basically get a massive unreadable blob of red next to a massive unreadable green blog.

When mucking around with the fixtures I found those tests qutie difficult to parse if there was a failure. This one at least had a separate row I could check to see if adding a new column had been added quickly.

But perhaps that's just me!

Reply:
👍 Understood, that does make sense!